controllers/competition_supervisor/competition_supervisor.py

```diff
@@ -36,6 +36,21 @@ def record_animation(supervisor: Supervisor, file_path: Path) -> Iterator[None]:
     yield
     supervisor.animationStopRecording()
 
+@contextlib.contextmanager
+def record_video(supervisor: Supervisor, file_path: Path) -> Iterator[None]:
+    file_path.parent.mkdir(parents=True, exist_ok=True)
+    print("Saving video to {}".format(file_path))
+    supervisor.movieStartRecording(
+        str(file_path),
+        width=1920,
+        height=1080,
+        quality=100,
+        codec=0,
+        acceleration=1,
+        caption=False)
```

Comment:
nit: local style is like
```suggestion
        caption=False,
    )
```